internal/clusterinfo/clusterinfo.go

```diff
@@ -0,0 +1,66 @@
+package clusterinfo
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	client "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterInfo is global information about where the ClusterLogForwarder is running.
+type ClusterInfo struct {
+	Version string // Version of the cluster.
+	ID      string // Unique identifier of the cluster.
+}
+
+// GetClusterInfo gets cluster info for the cluster we are running in.
+//
+// If is running in a Hosted Control Plane (hypershift management cluster)
+// this information describes the *guest* cluster, not the host cluster.
+// We assume in this case that CLF is running on behalf of the guest cluster to collect API audit logs.
+func Get(ctx context.Context, c client.Reader, cfg *rest.Config, namespace string) (*ClusterInfo, error) {
+	if hcp := getHCPInfo(ctx, cfg, namespace); hcp != nil { // Try HCP first
+		return hcp, nil
+	}
+	cv := &configv1.ClusterVersion{}
+	if err := c.Get(ctx, client.ObjectKey{Name: "version"}, cv); err != nil {
+		return nil, err
+	}
+	return &ClusterInfo{Version: cv.Spec.DesiredUpdate.Version, ID: string(cv.Spec.ClusterID)}, nil
+}
+
+var (
+	hcpGVR = schema.GroupVersionResource{
+		Resource: "hostedcontrolplanes",
+		Group:    "hypershift.openshift.io",
+		Version:  "v1beta1",
+	}
+	hcpGVK = schema.GroupVersionKind{
+		Kind:    "HostedControlPlane",
+		Group:   "hypershift.openshift.io",
+		Version: "v1beta1",
+	}
+)
+
+// getHCPInfo returns ClusterInfo from a hypershift control plane if found, nil otherwise.
+func getHCPInfo(ctx context.Context, cfg *rest.Config, namespace string) *ClusterInfo {
+	// TODO: using dynamic client because of dependency problems with the hypershift API package:
+	// When https://issues.redhat.com/browse/HOSTEDCP-336 is fixed we can enable clusterinfo_static.go
+	dc, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return nil
+	}
+	l, err := dc.Resource(hcpGVR).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil || len(l.Items) != 1 {
```

Comment:
I expect it won't but do we care if it does not equal 1?  Don't we just car that it is greater then zero?

Reply:
There should never be more than one, so it doesn't really matter. 
My thinking is if anything does not look as expected for the HCP case, then we fall back to the original behaviour. Multiple HCPs is not expected, and we don'tt know which one we should use.